include/roboteam_ai/stp/Play.hpp

```diff
@@ -5,62 +5,106 @@
+
+    /**
+     * Gets the score for the current play
+     *
+     * On the contrary to isValidPlay() this checks how good the play actually is
+     * return in range of 0 - 100
+     *
+     * @param world World to get the score for (world_new::World::instance())
+     * @return The score, 0 - 100
+     */
+    [[nodiscard]] virtual uint8_t score(world_new::World* world) noexcept = 0;
+
+    /**
+     * Virtual default dtor, ensures proper destruction of Play
+     */
+    virtual ~Play() = default;
+
+    /**
+     * Default ctor, ensures proper construction of Play
+     */
+    Play() = default;
+
     /**
-     * Play class that's used in the STP model
-     * on update traverses every Role, and updates it.
+     * Default move-ctor, ensures proper move-construction of Play
      */
-    class Play {
-    public:
-        constexpr static size_t ROBOT_COUNT = 11;
-
-        /**
-         * Updates all the roles
-         * @param info Information to pass down to the Roles
-         * @return Status depending on return value,
-         * if all finished -> finished
-         * if any failed -> failed
-         * if any waiting -> waiting
-         * otherwise -> running
-         */
-        [[nodiscard]] virtual Status update(TacticInfo const& info) noexcept;
-
-        /**
-         * Checks whether the current play is a valid play
-         * @param world World to check for (world_new::World::instance())
-         * @return true if valid, false if not
-         */
-        [[nodiscard]] virtual bool isValidPlay(world_new::World* world) noexcept = 0;
-
-        /**
-         * Gets the score for the current play
-         *
-         * On the contrary to isValidPlay() this checks how good the play actually is
-         * return in range of 0 - 100
-         *
-         * @param world World to get the score for (world_new::World::instance())
-         * @return The score, 0 - 100
-         */
-        [[nodiscard]] virtual uint8_t score(world_new::World* world) noexcept = 0;
-
-        /**
-         * Virtual default ctor, ensures proper destruction of Play
-         */
-        virtual ~Play() = default;
-
-    protected:
-        /**
-         * The roles, probably constructed in the parameter
-         */
-        std::array<std::unique_ptr<Role>, ROBOT_COUNT> roles;
-    };
-
-} // namespace rtt::ai::stp
-
-#endif //RTT_PLAY_HPP
+    Play(Play&& other) = default;
+
+   protected:
+    /**
+     * The roles, constructed in ctor of a play
+     */
+    std::array<std::unique_ptr<Role>, ROBOT_COUNT> roles;
+
+    /**
+     * The tacticInfos, constructed in assignRoles
+     */
+    std::unordered_map<std::string, StpInfo> stpInfos;
+
+    /**
+     * The world
+     */
+    rtt::world_new::World* world;
+
+    /**
+     * The Field
+     */
+    rtt::ai::Field field;
+
+   private:
+    /**
+     * Assigns robots to roles
+     */
+    virtual void assignRoles() noexcept = 0;
```

Comment:
Protected, right? You will want to override this method

```suggestion
   protected:
    /**
     * Assigns robots to roles
     */
    virtual void assignRoles() noexcept = 0;
```